asyncio increase throughput

000Justin000 opened the pull request:
Summary:
This diffs add a check in the fetcher, that if the dataset to be fetched has a function "getitems" then use it for fetching a batch of elements, as oppose to one by one. This is benefical for io bounded usage.

Differential Revision: D38876984

soumith reviewed (changes requested):
why `getitems`? has this API been discussed with anyone? cc: @vitaly-fedyunin

Commit:
asyncio increase throughput (#83786)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/83786

This diffs add a check in the fetcher, that if the dataset to be fetched has a function "getitems" then use it for fetching a batch of elements, as oppose to one by one. This is benefical for io bounded usage.

Differential Revision: D38876984

fbshipit-source-id: ec83d4ce3cd944e7da4a9b0645664fa00a5969bf
